esi_leap/api/controllers/v1/utils.py

```diff
@@ -22,3 +24,14 @@ def verify_admin(cdict):
 
     except oslo_policy.policy.PolicyNotAuthorized:
         return False
+
+
+def verify_resource_ownership(cdict, offer_dict):
```

Comment:
I think you probably just want to pass in `resource_object` and `project_id` instead of `offer_dict`; it's hard to understand what needs to be in `offer_dict`.